rust/sdk-processor/src/steps/parquet_default_processor/size_buffer.rs

```diff
@@ -0,0 +1,260 @@
+use crate::{
+    parquet_processors::{ParquetTypeEnum, ParquetTypeStructs},
+    steps::parquet_default_processor::gcs_handler::Uploadable,
+};
+use anyhow::Result;
+use aptos_indexer_processor_sdk::{
+    traits::{
+        pollable_async_step::PollableAsyncRunType, NamedStep, PollableAsyncStep, Processable,
+    },
+    types::transaction_context::{TransactionContext, TransactionMetadata},
+    utils::errors::ProcessorError,
+};
+use async_trait::async_trait;
+use std::{collections::HashMap, time::Duration};
+use tracing::debug;
+
+struct ParquetBuffer {
+    pub buffer: ParquetTypeStructs,
+    pub buffer_size_bytes: usize,
+    current_batch_metadata: TransactionMetadata,
+}
+
+impl ParquetBuffer {
+    fn new(parquet_type: &ParquetTypeEnum) -> Self {
+        Self {
+            buffer: ParquetTypeStructs::default_for_type(parquet_type),
+            buffer_size_bytes: 0,
+            current_batch_metadata: TransactionMetadata::default(),
```

Comment:
Setting this to the default could introduce potentially incorrect metadata to version tracker. Should we set this to None instead if it hasn't processed anything yet?

Reply:
yeah I think it makes sense to just reset to None after cleanup or None before any upload